### DISCUSSION TITLE
Inquiry Regarding 'ShouldQueueOnMainThread' and GATT Status 133 on Samsung Galaxy S20 and above

### DISCUSSION THREAD

**Discussion by 92-Theo** [dotnet-bluetooth-le/dotnet-bluetooth-le#745] [Q&A] [answered]
Hello,

I've been working with your BLE library for Xamarin, and it's been incredibly helpful. I have a couple of questions that I'd appreciate your insight on:

1. Regarding 'ShouldQueueOnMainThread': Could you please explain the rationale behind adding the 'ShouldQueueOnMainThread' field in the BleImplementation class? I'm particularly interested in understanding the problems it was designed to solve or any performance improvements it offers.
2. GATT Status Code 133 on Samsung Galaxy S20 and Above: I've noticed that on Samsung Galaxy S20 and newer models, GATT status code 133 occasionally occurs. When this happens, it renders all GATT-related apps on Android non-functional until the system is rebooted. Does the 'ShouldQueueOnMainThread' option have any relation to this issue, or is it designed to alleviate such problems? If not, do you have any suggestions on how to handle this?

I appreciate any guidance you can provide on these issues.

Best regards,

**Comment by ygl-rg** [comment]
I had a similar issue with the samsung M34. Once gatt exception is thrown, The app has to be restarted otherwise the ble will never been working.

**Comment by axa88** [comment, marked as answer]
>     1. Regarding 'ShouldQueueOnMainThread': Could you please explain the rationale behind adding the 'ShouldQueueOnMainThread' field in the BleImplementation class? I'm particularly interested in understanding the problems it was designed to solve or any performance improvements it offers.

late to the party but, while Ive not read this libraries source code but im pretty sure i know what you were inquiring about.
It was either a conscious design decision or an unconscious mistake, but Android needs to have certain processes happen on the Main Ui Thread (as well as made synchronously)
This field I would imagine is just a easy way to turn this functionality on or off if Android ever figures out this was a mistake and fixes it.

If you hadn't already you can read more about it [here](https://punchthrough.com/android-ble-guide/)
